lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -112,6 +112,17 @@ class AwsCompileFunctions {
         throw new Error(`No artifact path is set for function: ${functionName}`);
       }
 
+      if (this.serverless.service.provider.deploymentBucket) {
+        const deploymentBucket = this.serverless.service.provider.deploymentBucket;
```

Comment:
also we should use the `getServerlessDeploymentBucketName()` you updated here imo to make use of the region validation logic.